allennlp/common/plugins.py

```diff
@@ -1,52 +1,23 @@
 """
 Plugin management.
 
-AllenNLP supports loading "plugins" dynamically. A plugin is just a Python package that is found
-by AllenNLP with the methods provided in this module.
-
-There are two ways of declaring plugins for discovery:
-
-- Writing the package name to import in a file (typically ".allennlp_plugins", in the current
-  directory from which the command "allennlp" is run). This is the simplest approach.
-
-- Creating a folder called "allennlp_plugins" that's in the Python path when you run the
-  "allennlp" command (typically under your project's root directory), then creating a subfolder
-  with the name you want and creating an "__init__.py" file that imports the code you want (e.g.,
-  your Python package). This option is preferred when you want to create a pip-installable
-  package and you want to make your AllenNLP plugin available when users install your package.
-  See [allennlp-server](https://github.com/allenai/allennlp-server) for an example.
+AllenNLP supports loading "plugins" dynamically. A plugin is just a Python package that
+can be found and imported by AllenNLP. This is done by creating a file named `.allennlp_plugins`
+in the directory where the `allennlp` command is run that lists the modules that should be loaded,
+one per line.
 """
+
 import importlib
 import logging
 import os
-import pkgutil
-import sys
 from typing import Iterable
 
-from allennlp.common.util import push_python_path
+from allennlp.common.util import push_python_path, import_module_and_submodules
 
 logger = logging.getLogger(__name__)
 
 
-def discover_namespace_plugins(
-    namespace_name: str = "allennlp_plugins",
-) -> Iterable[pkgutil.ModuleInfo]:
-    """
-    Returns an iterable of the plugins found, declared within the namespace package `namespace_name`.
-    """
-    try:
-        reload = namespace_name in sys.modules
-
-        namespace_module = importlib.import_module(namespace_name)
-
-        if reload:
-            importlib.reload(namespace_module)
-
-        return pkgutil.iter_modules(
-            namespace_module.__path__, namespace_module.__name__ + "."  # type: ignore
-        )
-    except ModuleNotFoundError:
-        return []
+DEFAULT_PLUGINS = ("allennlp_models",)
```

Comment:
Why is this a tuple? Isn't that more of a list kind of thing?

Reply:
I just think tuples are better than lists for globals like this since tuples are immutable and take less memory